### PR TITLE
Issue 2772253: Add link forgotten password in error message in Login checkout pane

### PR DESCRIPTION
**Pull request by aristova**
Issue
https://www.drupal.org/node/2772253

### CODE REVIEW

File: modules/checkout/src/Plugin/Commerce/CheckoutPane/Login.php [hunk 1]
```diff
@@ -243,8 +242,11 @@ public function validatePaneForm(array &$pane_form, FormStateInterface $form_sta
     $values = $form_state->getValue($pane_form['#parents']);
     $username = $values['returning_customer']['name'];
     $password = trim($values['returning_customer']['password']);
+    // Setting url for "forgotten password" link in the error message.
+    $route_parameters = !empty($username) ? array('name' => $username) : array();
```
Comment (borisson) [inline modules/checkout/src/Plugin/Commerce/CheckoutPane/Login.php]: With all other changes lines in this PR are short array syntax, but this line uses long syntax, let's change that as well.

File: modules/checkout/src/Plugin/Commerce/CheckoutPane/Login.php [hunk 2]
```diff
@@ -243,8 +242,11 @@ public function validatePaneForm(array &$pane_form, FormStateInterface $form_sta
     $values = $form_state->getValue($pane_form['#parents']);
     $username = $values['returning_customer']['name'];
     $password = trim($values['returning_customer']['password']);
+    // Setting url for "forgotten password" link in the error message.
+    $route_parameters = !empty($username) ? ['name' => $username] : [];
```
Comment (bojanz) [inline modules/checkout/src/Plugin/Commerce/CheckoutPane/Login.php:246]: $route_parameters should be $query. The real route parameters are the empty array passed to Url.

File: modules/checkout/src/Plugin/Commerce/CheckoutPane/Login.php [hunk 3]
```diff
@@ -243,8 +242,11 @@ public function validatePaneForm(array &$pane_form, FormStateInterface $form_sta
...
+    $route = Url::fromRoute('user.pass', [], ['query' => $route_parameters])->toString();
```
Comment (bojanz) [inline modules/checkout/src/Plugin/Commerce/CheckoutPane/Login.php:247]: $route should be $password_url, since we're not getting a route.

**Comment by bojanz** [comment]
Addressed my own feedback, committed in 283dd88ec7b91fdedbf2913b681e6036370bc31f. Thanks!

**Pull request closed without merging**